### PR TITLE
external/test: compare the performance of external writer

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close #xxx

Problem Summary:

### What is changed and how it works?

Add a simple unit test framework that wants to check the throughput, CPU profile and heap profile of external reader/writer and their compositions.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

```
$ go test ./br/pkg/lightning/backend/external -v --tags=intest -test.run TestCompare --testing-storage-uri "s3://nfs/tianyuan-ut?access-key=xxx&secret-access-key=xxx&endpoint=https://xxx&force-path-style=false&region=xxx&provider=ks"
...
    bench_test.go:210: base speed for 1200000000 bytes: 14.68 MB/s
...
    bench_test.go:215: writer speed for 1200000000 bytes: 54.18 MB/s
```

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/backend/external/bench_test.go [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+	if s.beforeWriterClose != nil {
+		s.beforeWriterClose()
+	}
+	require.NoError(s.t, writer.Close(ctx))
+	if s.afterWriterClose != nil {
+		s.afterWriterClose()
+	}
+}
+
+func TestCompare(t *testing.T) {
```
Comment (tangenta) [inline br/pkg/lightning/backend/external/bench_test.go:162]: Use `BenchmarkXXX` instead?

Comment (lance6716) [reply br/pkg/lightning/backend/external/bench_test.go:162]: I have used t.Skip() in line 33. Another reason is that I want to do more calculation in the tests, like compute the throughput ratio of external write / direct writing, and output profile files only for part of code. Benchmark will not help much about these purpose.

**Comment by lance6716** [comment]
/hold

from the CPU profile I see require.NoError costs too much CPU, I will change it to panic

**Review by ywqzzy — APPROVED** [review]

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**